doc/source/data/api/data_representations.rst

```diff
@@ -3,54 +3,53 @@
 Data Representations
 ====================
```

Comment:
Aren't these internal APIs?

Reply:
> Aren't these internal APIs?

They are internal. I feel we should remove them (this PR is mostly refactoring stuff). cc @matthewdeng.

Reply:
Some of these APIs are Public or Developer APIs. cc @clarkzinzow

Reply:
The block/batch types are `DeveloperAPI`, the `TableRow` type is `PublicAPI`, and the tensor extension types are `PublicAPI`.

Reply:
Cool, then let's keep this file.